cbor-schema/plt.cddl

```diff
@@ -175,7 +186,7 @@ token-module-state = {
     ; The name of the token
     "name": text,
     ; A URL pointing to the token metadata
-    "metadata": text,
+    "metadata": metadata-url,
```

Comment:
I am not 100% sure we should use a map for this, rather than having an additional key e.g. "metadata-sha256". This might be more future compatible if we want to use different hashing schemes in future (for instance, if someone breaks SHA-256).

Reply:
Could also provide a hashing scheme with the `hash`, i.e.

```
hash = {
  "value": bytes .size(32),
  "scheme": "sha256",
}

metadata-url = {
  "url": text,
  "checksum": hash,
}
```

What do you think?

Reply:
We should probably not assume that the checksum is 32 bytes, if we want to be future-proof.
I like the current approach, since we can easily extend with more/different checksums for the metadata url.

Reply:
I feel like we should either be specific from the start:

```
sha256-hash = bytes .size(32)
sha512-hash = bytes .size(64)

metadata-url = { 
  "url": text, 
  ? "checksum-sha256": sha256-hash, 
  ? "checksum-sha512": sha512-hash, ; added down the line when needed
  ...
}
```

Or add to a union:

```
hash = sha256-hash / sha512-hash

sha256-hash = {
  "value": bytes .size(32),
  "scheme": "sha256",
}

sha512-hash = {
  "value": bytes .size(64),
  "scheme": "sha512",
}

metadata-url = {
  "url": text,
  ? "checksum": hash,
}
```

I'm leaning towards the latter, as I feel like it's easier to work with at the consumer level. I don't have a strong preference though. Maybe you can elaborate on why you don't like a map approach @td202 ?

Reply:
After discussion with @td202 we landed on:

```
; Type alias for a sha256 hash
sha256-hash = bytes .size(32)

; Type alias for metadata with a URL and checksum
metadata-url = {
    ; A string field representing the URL
    "url": text,
    ; An optional sha256 checksum value tied to the content of the URL
    ? "checksumSha256": sha256-hash
    ; Additional fields may be included for future extensibility, e.g. another hash algorithm.
    * text => any
}
```